cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,297 @@
+		i18n.G("Set"),
+		"Snap",
+		i18n.G("Age"),
+		i18n.G("Version"),
+		// TRANSLATORS: 'Rev' is an abbreviation of 'Revision'
+		i18n.G("Rev"),
+		i18n.G("Size"),
+		// TRANSLATORS: 'Notes' as in 'Comments'
+		i18n.G("Notes"))
+	for _, sg := range list {
+		for _, sh := range sg.Snapshots {
+			note := "-"
+			if sh.Broken != "" {
+				note = "broken: " + sh.Broken
+			}
+			size := quantity.FormatAmount(uint64(sh.Size), -1) + "B"
+			fmt.Fprintf(w, "%d\t%s\t%s\t%s\t%s\t%s\t%s\n", sg.ID, sh.Snap, x.fmtDuration(sh.Time), sh.Version, sh.Revision, size, note)
+		}
+	}
+	return nil
+}
+
+type saveCmd struct {
+	waitMixin
+	durationMixin
+	Users      string `long:"users"`
+	Positional struct {
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes"`
+}
+
+var shortSaveHelp = i18n.G("Save a snapshot of the current data")
+var longSaveHelp = i18n.G(`
+The save command creates a snapshot of the current data for the given snaps.
+`)
+
+func (x *saveCmd) Execute([]string) error {
+	var users []string
+	if len(x.Users) > 0 {
+		users = strings.Split(x.Users, ",")
+	}
+	setID, changeID, err := x.client.SnapshotMany(installedSnapNames(x.Positional.Snaps), users)
+	if err != nil {
+		return err
+	}
+	if _, err := x.wait(changeID); err != nil {
+		if err == noWait {
+			return nil
+		}
+		return err
+	}
+
+	y := &savedCmd{
+		clientMixin:   x.clientMixin,
+		durationMixin: x.durationMixin,
+		ID:            snapshotID(setID),
+	}
+	return y.Execute(nil)
+}
+
+type forgetCmd struct {
+	waitMixin
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortForgetHelp = i18n.G("Delete a snapshot")
+var longForgetHelp = i18n.G(`
+The forget command deletes a snapshot.
+`)
+
+func (x *forgetCmd) Execute([]string) error {
+	snaps := installedSnapNames(x.Positional.Snaps)
+	changeID, err := x.client.ForgetSnapshots(uint64(x.Positional.ID), snaps)
+	if err != nil {
+		return err
+	}
+	_, err = x.wait(changeID)
+	if err == noWait {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) > 0 {
+		// TRANSLATORS: the %s is a comma-separated list of quoted snap names
+		fmt.Fprintf(Stdout, i18n.NG("Snapshot #%d of snap %s forgotten.\n", "Snapshot #%d of snaps %s forgotten.\n", len(snaps)), x.Positional.ID, strutil.Quoted(snaps))
+	} else {
+		fmt.Fprintf(Stdout, i18n.G("Snapshot #%d forgotten.\n"), x.Positional.ID)
+	}
+	return nil
+}
+
+type checkSnapshotCmd struct {
+	waitMixin
+	Users      string `long:"users"`
+	Positional struct {
+		ID    snapshotID          `positional-arg-name:"<id>"`
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortCheckHelp = i18n.G("Check a snapshot")
+var longCheckHelp = i18n.G(`
+The check-snapshot command checks a snapshot against its hashsums.
```

Comment:
Maybe something more user friendly, `The check snapshot command verifies correctness of given snapshot`, or `.. consistency..` although that feels a bit technical.